core/common/src/main/java/alluxio/conf/InstancedConfiguration.java

```diff
@@ -242,97 +249,47 @@ public void merge(Map<?, ?> properties, Source source) {
-    String rawValue = get(key);
-
-    try {
-      return Integer.parseInt(rawValue);
-    } catch (NumberFormatException e) {
-      throw new RuntimeException(ExceptionMessage.KEY_NOT_INTEGER.getMessage(rawValue, key));
-    }
+  public String getString(PropertyKey key) {
+    return (String) get(key);
   }
 
   @Override
-  public long getLong(PropertyKey key) {
-    String rawValue = get(key);
-
-    try {
-      return Long.parseLong(rawValue);
-    } catch (NumberFormatException e) {
-      throw new RuntimeException(ExceptionMessage.KEY_NOT_LONG.getMessage(rawValue, key));
-    }
+  public int getInt(PropertyKey key)
+  {
+    return (Integer) get(key);
   }
 
   @Override
   public double getDouble(PropertyKey key) {
-    String rawValue = get(key);
-
-    try {
-      return Double.parseDouble(rawValue);
-    } catch (NumberFormatException e) {
-      throw new RuntimeException(ExceptionMessage.KEY_NOT_DOUBLE.getMessage(rawValue, key));
-    }
-  }
-
-  @Override
-  public float getFloat(PropertyKey key) {
-    String rawValue = get(key);
-
-    try {
-      return Float.parseFloat(rawValue);
-    } catch (NumberFormatException e) {
-      throw new RuntimeException(ExceptionMessage.KEY_NOT_FLOAT.getMessage(rawValue, key));
-    }
+    return (Double) get(key);
   }
 
   @Override
   public boolean getBoolean(PropertyKey key) {
-    String rawValue = get(key);
-    try {
-      return FormatUtils.parseBoolean(rawValue);
-    } catch (Exception e) {
-      throw new RuntimeException(ExceptionMessage.KEY_NOT_BOOLEAN.getMessage(rawValue, key));
-    }
+    Object rawValue = get(key);
+    return (Boolean) rawValue;
   }
 
   @Override
-  public List<String> getList(PropertyKey key, String delimiter) {
-    Preconditions.checkArgument(delimiter != null,
-        "Illegal separator for Alluxio properties as list");
-    String rawValue = get(key);
-    return ConfigurationUtils.parseAsList(rawValue, delimiter);
+  public List<String> getList(PropertyKey key) {
+    String value = (String) get(key);
+    return ConfigurationUtils.parseAsList(value, key.getDelimiter());
   }
 
   @Override
   public <T extends Enum<T>> T getEnum(PropertyKey key, Class<T> enumType) {
-    String rawValue = get(key).toUpperCase();
-    try {
-      return Enum.valueOf(enumType, rawValue);
-    } catch (IllegalArgumentException e) {
-      throw new RuntimeException(ExceptionMessage.UNKNOWN_ENUM.getMessage(rawValue, key,
-          Arrays.toString(enumType.getEnumConstants())));
-    }
+    checkArgument(key.getEnumType().equals(enumType), "PropertyKey %s is not of enum type", key);
+    return enumType.cast(get(key));
   }
 
   @Override
   public long getBytes(PropertyKey key) {
-    String rawValue = get(key);
-
-    try {
-      return FormatUtils.parseSpaceSize(rawValue);
-    } catch (Exception ex) {
-      throw new RuntimeException(ExceptionMessage.KEY_NOT_BYTES.getMessage(rawValue, key));
-    }
+    return (Long) get(key);
```

Comment:
So we're skipping the FormatUtils.parseSpaceSize,  will it work for something like '1GB'?